### ISSUE TITLE
Issue with custom blocks with multi-input slots while editing them

### ISSUE THREAD

**Issue by HM100** [jmoenig/Snap#2702]
Found a bug, this time while editing a custom block's name or translating a custom block into another language if it has multi-input slots. This causes the content put inside the inputs (Even replacing the multi input with a variable):

Here's a screenshot prior to block's edit:
![image](https://user-images.githubusercontent.com/30199227/95662761-f2401d80-0b41-11eb-9951-bcfa0b50037f.png)

Here's a screenshot after changing a block's name (Note that the multi-input slot (The last one) is reverted back to its non-modified form aka every content put into the input slots was wiped out):
![image](https://user-images.githubusercontent.com/30199227/95662809-3a5f4000-0b42-11eb-9785-ed0e1090d6a6.png)

Expected Result:
- Renaming a block's spec (Such as adding new inputs) or translating the block into another language shouldn't touch any multi-input slots content

Actual Result:
- Renaming a block's spec (Such as adding new inputs) or translating the block into another language wipes out every content put into the multi-input slot, breaking up any other script that uses the block.

Pics say it all

**Comment by jmoenig** [comment]
yeah, that's a known loose end in variadic input slots. The good part about it is that I didn't yet write any code about it, so it's not a "bug" in that it does something unexpectedly "wrong" as opposed to what the code claims it does, but it is very much a bug in that it contradicts user expectations :-)

**Comment by HM100** [comment]
Yeah! This can break projects that use such kind of blocks but then they want to either edit the block name or translate it into another language (Such as German)

**Comment by HM100** [comment]
Sorry for necroposting but appears that in the latest snap! version, I translated a custom ``average %n...`` block that has a mutli-slot of numerical inputs into German and it didn't broke any existing scripts. Seems that it has been fixed. I'm closing that now but if one finds it again, don't hesitate to reopen it.
